.travis.yml

```diff
@@ -16,19 +16,24 @@ osx_setup: &osx_setup
       packages:
         - openssl
   env:
-    - >
+    - &env_pyenv >
       PYENV_ROOT="${HOME}/.pyenv"
       PATH="${PYENV_ROOT}/shims:${PATH}"
+    - >
       # These flags are necessary to get OpenSSL working. See
       # https://github.com/pyenv/pyenv/wiki/Common-build-problems#error-the-python-ssl-extension-was-not-compiled-missing-the-openssl-lib.
       PATH="/usr/local/opt/openssl/bin:$PATH"
       LDFLAGS="-L/usr/local/opt/openssl/lib"
       CPPFLAGS="-I/usr/local/opt/openssl/include"
   before_install:
-    - >
+    - &pyenv_install >
       git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
       && ${PYENV_ROOT}/bin/pyenv install 3.6.8
       && ${PYENV_ROOT}/bin/pyenv global 3.6.8
+  before_script:
+    # Override file handler and thread limits
```

Comment:
Are we seeing a failure without these? Should we set them consistently across platforms?

Reply:
Yes, master sometimes would fail with the `too many open files error`: https://travis-ci.org/pantsbuild/setup/jobs/505521244#L283

However, we didn't catch this because of the missing `check=True` in `subprocess.run()`, meaning the subprocess would fail but not cause the entire `ci.py` script to fail like we would want it to.

--

No idea if we should add it to Linux shards too. Note we do not do this in Pants CI for Linux; we override for every OSX shard and for Linux shards that run Rust tests, but not for our normal Linux shards.

I'm fine with adding it, although would err on the side of making as few changes as necessary to get green CI.